### ISSUE TITLE
Multiple UseQuery Hooks in Next.js Default Store Front Causing Backend Errors and Unresponsive Requests

### ISSUE THREAD

**Issue by TheBlankness** [medusajs/medusa#3874]
**Issue Description:**
In our Next.js default store front application, we are using multiple UseQuery hooks to fetch data from the backend API. However, we are facing backend errors and subsequent requests are not being responded to even though the backend appears to be running normally.

**Error Logs:**
The backend logs show that the application has gracefully stopped, but it is still running as usual. However, it is not responding to subsequent requests from the frontend.

**Expected Behavior:**
The application should be able to handle multiple UseQuery hooks in the Next.js default store front without causing backend errors and unresponsive requests.

**Steps to Reproduce:**

Set up a Next.js default store front application.
Use multiple UseQuery hooks to fetch data from the backend API.
Trigger requests from the frontend.
Observe backend error logs and unresponsive requests.
**Actual Behavior:**
Backend errors occur and subsequent requests from the frontend are not being responded to, even though the backend appears to be running normally.

Additional Information:

"@medusajs/medusa": "^1.8.0",
"next": "^12.2.0",
"@medusajs/medusa-js": "^1.3.7",
 "@tanstack/react-query": "^4.22.4",

**Comment by chemicalkosek** [comment]
Can you update medusa to 1.8.2?

**Comment by TheBlankness** [comment]
updated to 1.8.2 seems to be the same. tried with and without reverse proxy. same issue

**Comment by tabilzad** [comment]
+1 also on 1.8.2

**Comment by olivermrbl** [comment]
There are currently two known problems, we are trying to tackle, that could cause these issues.
1. After having upgraded Typeorm, we've seen that a range of our APIs have started generating very large database queries, that ultimately lead to an out-of-memory error. The issues spawn because we switched the query strategy of Typeorm, such that instead of doing separate queries when fetching relations, it does one large query with a bunch of joins. We've solved the issue in our more "query-heavy" APIs (Products, Carts, Orders) in 1.9 + 1.10.
2. We've seen that using the inventory and stock location module can drain the number of database connections which ultimately leads to an unresponsive application. It is yet to be solved across the entire API, but we are working hard on a couple of potential solutions that will hopefully eliminate the issue for good.

Can I get all of you to upgrade to our latest version and get back with an update? That would narrow down the place where these issues could spawn from.
